### PR TITLE
cmd/tailscale: combine serve and funnel for debug wip funnel stream model

### PR DESCRIPTION
**Pull request by tylersmalley**
> **Note**
> Behind the `TAILSCALE_USE_WIP_CODE` flag

In preparing for incoming CLI changes, this PR merges the code path for the `serve` and `funnel` subcommands.

See the [parent issue](https://github.com/tailscale/tailscale/issues/8489) for more context.

The following commands will run in foreground mode when using the environment flag. 

```
./tool/go run -exec "sudo env TAILSCALE_USE_WIP_CODE=1" ./cmd/tailscaled --socket=/tmp/ts/ts1.sock --state=/tmp/ts/ts1.state --statedir=/Library/Tailscale

TAILSCALE_USE_WIP_CODE=1 ./tool/go run ./cmd/tailscale --socket=/tmp/ts/ts1.sock serve localhost:3000
TAILSCALE_USE_WIP_CODE=1 ./tool/go run ./cmd/tailscale --socket=/tmp/ts/ts1.sock funnel localhost:3000
```

Replaces #9134
Updates #8489

### CODE REVIEW

**Review by bradfitz — DISMISSED** [review]
I want to agree on the plan before this goes in. This commit message is super short on any details of what's going on and why.

**Comment by tylersmalley** [comment]
@bradfitz I have updated the commit message.

I am happy to discuss the overall product goals with the changes, or any issues you might have with them. They are outlined in a [design document](https://docs.google.com/document/d/180hvH5D1l0381X4SdQYUKneRGMvfGP3irzb0Vuowgxk/edit) and [linear](https://linear.app/tailscale/issue/ENG-2054/funnel-cli-evolution).

**Comment by tylersmalley** [comment]
@bradfitz are you OK with this change after our discussion?

File: cmd/tailscale/cli/serve.go [hunk 1]
```diff
@@ -32,7 +32,16 @@ import (
 	"tailscale.com/version"
 )
 
-var serveCmd = newServeCommand(&serveEnv{lc: &localClient})
+var serveCmd = func() *ffcli.Command {
+	se := &serveEnv{lc: &localClient}
+	// This flag is used to switch to an in-development
+	// implementation of the tailscale funnel command.
+	// See https://github.com/tailscale/tailscale/issues/7844
+	if os.Getenv("TAILSCALE_FUNNEL_DEV") == "on" {
```
Comment (willnorris) [inline cmd/tailscale/cli/serve.go:40]: consider using `envknob.UseWIPCode` that you found ?

**Review by willnorris — APPROVED** [review]
largely just refactoring, so LGTM.  From our discussion, sounds like most of Brad's immediate concerns have been addressed, but also this is still behind a flag and designs are continuing to iterate. 👍🏻

File: cmd/tailscale/cli/serve.go (hunk 1 above)
Comment (tylersmalley) [reply cmd/tailscale/cli/serve.go]: 👍 updated in f128e97331367222543dd9f26c1784f1dd43bda2

**Review by bradfitz — COMMENTED** [review]
(removing request changes)

**Pull request merged**